ksqldb-execution/src/test/java/io/confluent/ksql/execution/builder/KsqlQueryBuilderTest.java

```diff
@@ -234,39 +241,44 @@ public void shouldBuildValueSerde() {
   }
 
   @Test
-  public void shouldTrackSchemasUsed() {
+  public void shouldTrackKeySchemasUsed() {
     // When:
-    ksqlQueryBuilder.buildValueSerde(
+    ksqlQueryBuilder.buildKeySerde(
         FORMAT_INFO,
-        SOME_SCHEMA,
+        PHYSICAL_SCHEMA,
         queryContext
     );
 
     // Then:
-    final Map<String, PhysicalSchema> schemas = ksqlQueryBuilder.getSchemas().getSchemas();
+    final Map<String, SchemaInfo> schemas = ksqlQueryBuilder.getSchemas().getSchemas();
     assertThat(schemas.entrySet(), hasSize(1));
-    assertThat(schemas.get("fred.context"), is(SOME_SCHEMA));
+    assertThat(schemas.get("fred.context"), is(new SchemaInfo(
+        LOGICAL_SCHEMA,
+        Optional.of(KeyFormat.nonWindowed(
+            FormatInfo.of("AVRO", ImmutableMap.of("fullSchemaName", "io.confluent.ksql")),
+            SerdeFeatures.of(SerdeFeature.UNWRAP_SINGLES))),
+        Optional.empty()
+    )));
   }
 
   @Test
-  public void shouldTrackSchemasTakingIntoAccountSerdeFeatures() {
-    // Given:
-    final PhysicalSchema schema = PhysicalSchema.from(
-        SOME_SCHEMA.logicalSchema(),
-        SerdeFeatures.of(),
-        SerdeFeatures.of(SerdeFeature.UNWRAP_SINGLES)
-    );
-
+  public void shouldTrackValueSchemasUsed() {
```

Comment:
Is it worth adding a unit test to cover the `merge()` functionality required for tracking both key and value schema for the same logger prefix? I don't see coverage for that at the moment but I also don't feel too strongly since this code is essentially test-only.

Reply:
I'll add to the follow up PR.

Reply:
see https://github.com/confluentinc/ksql/pull/6366